src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -585,6 +585,31 @@ void StorageS3Configuration::addStructureAndFormatToArgsIfNeeded(
     }
 }
 
+void StorageS3Configuration::getTableFunctionArguments(ASTs & args) const
+{
+    if (!args.empty())
+    { /// Just check
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Arguments are not empty");
+    }
+
+    args.push_back(std::make_shared<ASTLiteral>(url.uri_str));
+    if (auth_settings[S3AuthSetting::no_sign_request])
+    {
+        args.push_back(std::make_shared<ASTLiteral>("NOSIGN"));
+    }
+    else
+    {
+        args.push_back(std::make_shared<ASTLiteral>(auth_settings[S3AuthSetting::access_key_id].value));
+        args.push_back(std::make_shared<ASTLiteral>(auth_settings[S3AuthSetting::secret_access_key].value));
+        if (!auth_settings[S3AuthSetting::session_token].value.empty())
+            args.push_back(std::make_shared<ASTLiteral>(auth_settings[S3AuthSetting::session_token].value));
+        if (format != "auto")
+            args.push_back(std::make_shared<ASTLiteral>(format));
+        if (!compression_method.empty())
```

Comment:
Is it ok not to add some other arguments like `structure` and `headers` as the docs suggest? https://clickhouse.com/docs/en/sql-reference/table-functions/s3

Reply:
Yes, this fields are already added in `addStructureAndFormatToArgsIfNeeded` methods.
https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/ObjectStorage/S3/Configuration.cpp#L399
These field required for table functions too, when access parameters need to add only for table engine case.

Reply:
Nitpick: perhaps add a comment saying this is already added somewhere else?